### PR TITLE
Fix silicon support

### PR DESCRIPTION
**Pull request by zbeucler2018**
Just had to reorder the platform conditionals to fix silicon support

### CODE REVIEW

**Review by pseudo-rnd-thoughts — CHANGES_REQUESTED** [review]
We don't have testing for macos, neither arm or intel. 
Could you add this

It should require modifying the `.github/workflows/test-python.yml` to use a strategy including the run-on cases
See this ALE example for it - https://github.com/Farama-Foundation/Arcade-Learning-Environment/blob/master/.github/workflows/ci.yml

**Comment by pseudo-rnd-thoughts** [comment]
Could you add to the wheel building workflow to build macos

**Comment by dej4vu** [comment]
### i install this commit successfully, but when i run the example bellow, it go wrong.    
Python 3.10.14
Macos 14.5 Apple M1 Pro 

$python3 ppo.py --game='Airstriker-Genesis'
>Using cpu device
2024-08-24 03:10:27.012 python3[35970:446285] +[NSXPCSharedListener endpointForReply:withListenerName:replyErrorCode:]: an error occurred while attempting to obtain endpoint for listener 'ClientCallsAuxiliary': Connection invalid
2024-08-24 03:10:27.013 python3[35970:446330] +[NSXPCSharedListener endpointForReply:withListenerName:replyErrorCode:]: an error occurred while attempting to obtain endpoint for listener 'com.apple.view-bridge': Connection invalid
2024-08-24 03:10:27.023 python3[35971:446286] +[NSXPCSharedListener endpointForReply:withListenerName:replyErrorCode:]: an error occurred while attempting to obtain endpoint for listener 'ClientCallsAuxiliary': Connection invalid
2024-08-24 03:10:27.023 python3[35972:446287] +[NSXPCSharedListener endpointForReply:withListenerName:replyErrorCode:]: an error occurred while attempting to obtain endpoint for listener 'ClientCallsAuxiliary': Connection invalid
2024-08-24 03:10:27.023 python3[35972:446359] +[NSXPCSharedListener endpointForReply:withListenerName:replyErrorCode:]: an error occurred while attempting to obtain endpoint for listener 'com.apple.view-bridge': Connection invalid
2024-08-24 03:10:27.024 python3[35971:446346] +[NSXPCSharedListener endpointForReply:withListenerName:replyErrorCode:]: an error occurred while attempting to obtain endpoint for listener 'com.apple.view-bridge': Connection invalid
2024-08-24 03:10:27.026 python3[35976:446291] +[NSXPCSharedListener endpointForReply:withListenerName:replyErrorCode:]: an error occurred while attempting to obtain endpoint for listener 'ClientCallsAuxiliary': Connection invalid
2024-08-24 03:10:27.028 python3[35976:446371] +[NSXPCSharedListener endpointForReply:withListenerName:replyErrorCode:]: an error occurred while attempting to obtain endpoint for listener 'com.apple.view-bridge': Connection invalid
2024-08-24 03:10:27.031 python3[35975:446290] +[NSXPCSharedListener endpointForReply:withListenerName:replyErrorCode:]: an error occurred while attempting to obtain endpoint for listener 'ClientCallsAuxiliary': Connection invalid
2024-08-24 03:10:27.031 python3[35975:446341] +[NSXPCSharedListener endpointForReply:withListenerName:replyErrorCode:]: an error occurred while attempting to obtain endpoint for listener 'com.apple.view-bridge': Connection invalid
2024-08-24 03:10:27.037 python3[35973:446288] +[NSXPCSharedListener endpointForReply:withListenerName:replyErrorCode:]: an error occurred while attempting to obtain endpoint for listener 'ClientCallsAuxiliary': Connection invalid
FALLBACK (log once): Fallback to SW vertex processing because buildPipelineState failed
FALLBACK (log once): Fallback to SW fragment processing because buildPipelineState failed
UNSUPPORTED (log once): UNEXPECTED/FATAL?: buildPipelineState failed to build fragment-fallback PSO, m_disable_code: 1001000
FALLBACK (log once): Fallback to SW vertex processing, m_disable_code: 1000
FALLBACK (log once): Fallback to SW fragment processing, m_disable_code: 1000000
FALLBACK (log once): Fallback to SW vertex processing because buildPipelineState failed
FALLBACK (log once): Fallback to SW fragment processing because buildPipelineState failed
UNSUPPORTED (log once): UNEXPECTED/FATAL?: buildPipelineState failed to build fragment-fallback PSO, m_disable_code: 1001000
FALLBACK (log once): Fallback to SW vertex processing, m_disable_code: 1000
FALLBACK (log once): Fallback to SW fragment processing, m_disable_code: 1000000
2024-08-24 03:10:27.042 python3[35974:446289] +[NSXPCSharedListener endpointForReply:withListenerName:replyErrorCode:]: an error occurred while attempting to obtain endpoint for listener 'ClientCallsAuxiliary': Connection invalid
2024-08-24 03:10:27.044 python3[35974:446436] +[NSXPCSharedListener endpointForReply:withListenerName:replyErrorCode:]: an error occurred while attempting to obtain endpoint for listener 'com.apple.view-bridge': Connection invalid
UNSUPPORTED (log once): Fallback to SW fragment processing with incorrect !(m_currentPrimitiveGroup(0x00000001) & (1U << mode(7))(0x00000080))
UNSUPPORTED (log once): Fallback to SW fragment processing with incorrect !(mode(7) == GL_POINTS)
FALLBACK (log once): Fallback to SW vertex processing because buildPipelineState failed
FALLBACK (log once): Fallback to SW fragment processing because buildPipelineState failed
UNSUPPORTED (log once): UNEXPECTED/FATAL?: buildPipelineState failed to build fragment-fallback PSO, m_disable_code: 1001000
FALLBACK (log once): Fallback to SW vertex processing, m_disable_code: 1000
FALLBACK (log once): Fallback to SW fragment processing, m_disable_code: 1000000
FALLBACK (log once): Fallback to SW vertex processing because buildPipelineState failed
FALLBACK (log once): Fallback to SW fragment processing because buildPipelineState failed
UNSUPPORTED (log once): UNEXPECTED/FATAL?: buildPipelineState failed to build fragment-fallback PSO, m_disable_code: 1001000
FALLBACK (log once): Fallback to SW vertex processing, m_disable_code: 1000
FALLBACK (log once): Fallback to SW fragment processing, m_disable_code: 1000000
FALLBACK (log once): Fallback to SW vertex processing because buildPipelineState failed
FALLBACK (log once): Fallback to SW fragment processing because buildPipelineState failed
UNSUPPORTED (log once): UNEXPECTED/FATAL?: buildPipelineState failed to build fragment-fallback PSO, m_disable_code: 1001000
FALLBACK (log once): Fallback to SW vertex processing, m_disable_code: 1000
FALLBACK (log once): Fallback to SW fragment processing, m_disable_code: 1000000
UNSUPPORTED (log once): Fallback to SW fragment processing with incorrect !(m_currentPrimitiveGroup(0x00000001) & (1U << mode(7))(0x00000080))
UNSUPPORTED (log once): Fallback to SW fragment processing with incorrect !(mode(7) == GL_POINTS)
FALLBACK (log once): Fallback to SW vertex processing because buildPipelineState failed
FALLBACK (log once): Fallback to SW fragment processing because buildPipelineState failed
UNSUPPORTED (log once): UNEXPECTED/FATAL?: buildPipelineState failed to build fragment-fallback PSO, m_disable_code: 1001000
FALLBACK (log once): Fallback to SW vertex processing, m_disable_code: 1000
FALLBACK (log once): Fallback to SW fragment processing, m_disable_code: 1000000
UNSUPPORTED (log once): Fallback to SW fragment processing with incorrect !(m_currentPrimitiveGroup(0x00000001) & (1U << mode(7))(0x00000080))
UNSUPPORTED (log once): Fallback to SW fragment processing with incorrect !(mode(7) == GL_POINTS)
UNSUPPORTED (log once): Fallback to SW fragment processing with incorrect !(m_currentPrimitiveGroup(0x00000001) & (1U << mode(7))(0x00000080))
UNSUPPORTED (log once): Fallback to SW fragment processing with incorrect !(mode(7) == GL_POINTS)
UNSUPPORTED (log once): Fallback to SW fragment processing with incorrect !(m_currentPrimitiveGroup(0x00000001) & (1U << mode(7))(0x00000080))
UNSUPPORTED (log once): Fallback to SW fragment processing with incorrect !(mode(7) == GL_POINTS)
UNSUPPORTED (log once): Fallback to SW fragment processing with incorrect !(m_currentPrimitiveGroup(0x00000001) & (1U << mode(7))(0x00000080))
UNSUPPORTED (log once): Fallback to SW fragment processing with incorrect !(mode(7) == GL_POINTS)
FALLBACK (log once): Fallback to SW vertex processing because buildPipelineState failed
FALLBACK (log once): Fallback to SW fragment processing because buildPipelineState failed
UNSUPPORTED (log once): UNEXPECTED/FATAL?: buildPipelineState failed to build fragment-fallback PSO, m_disable_code: 1001000
FALLBACK (log once): Fallback to SW vertex processing, m_disable_code: 1000
FALLBACK (log once): Fallback to SW fragment processing, m_disable_code: 1000000
UNSUPPORTED (log once): Fallback to SW fragment processing with incorrect !(m_currentPrimitiveGroup(0x00000001) & (1U << mode(7))(0x00000080))
UNSUPPORTED (log once): Fallback to SW fragment processing with incorrect !(mode(7) == GL_POINTS)
2024-08-24 03:10:27.103 python3[35969:446439] Connection Invalid error for service com.apple.hiservices-xpcservice.
2024-08-24 03:10:27.103 python3[35969:446343] Error received in message reply handler: Connection invalid
2024-08-24 03:10:27.105 python3[35969:446439] Error received in message reply handler: Connection invalid
2024-08-24 03:10:27.108 python3[35969:446284] +[NSXPCSharedListener endpointForReply:withListenerName:replyErrorCode:]: an error occurred while attempting to obtain endpoint for listener 'ClientCallsAuxiliary': Connection invalid
2024-08-24 03:10:27.108 python3[35969:446343] +[NSXPCSharedListener endpointForReply:withListenerName:replyErrorCode:]: an error occurred while attempting to obtain endpoint for listener 'com.apple.view-bridge': Connection invalid
FALLBACK (log once): Fallback to SW vertex processing because buildPipelineState failed
FALLBACK (log once): Fallback to SW fragment processing because buildPipelineState failed
UNSUPPORTED (log once): UNEXPECTED/FATAL?: buildPipelineState failed to build fragment-fallback PSO, m_disable_code: 1001000
FALLBACK (log once): Fallback to SW vertex processing, m_disable_code: 1000
FALLBACK (log once): Fallback to SW fragment processing, m_disable_code: 1000000
UNSUPPORTED (log once): Fallback to SW fragment processing with incorrect !(m_currentPrimitiveGroup(0x00000001) & (1U << mode(7))(0x00000080))
UNSUPPORTED (log once): Fallback to SW fragment processing with incorrect !(mode(7) == GL_POINTS)
-----------------------------
| time/              |      |
|    fps             | 175  |
|    iterations      | 1    |
|    time_elapsed    | 5    |
|    total_timesteps | 1024 |
-----------------------------
-------------------------------------------
| time/                   |               |
|    fps                  | 128           |
|    iterations           | 2             |
|    time_elapsed         | 15            |
|    total_timesteps      | 2048          |
| train/                  |               |
|    approx_kl            | 0.0021000488  |
|    clip_fraction        | 0.0759        |
|    clip_range           | 0.1           |
|    entropy_loss         | -8.32         |
|    explained_variance   | -0.0137217045 |
|    learning_rate        | 0.00025       |
|    loss                 | -0.0972       |
|    n_updates            | 4             |
|    policy_gradient_loss | -0.0102       |
|    value_loss           | 0.0205        |
-------------------------------------------

image
